ui/src/app/views/workspace-share/component.ts

```diff
@@ -95,13 +95,17 @@ export class WorkspaceShareComponent implements OnInit {
     }
     if (!this.usersLoading) {
       this.usersLoading = true;
-      const updateList = Array.from(this.workspace.userRoles);
-      updateList.push({
-        email: this.convertToEmail(this.toShare),
-        role: this.selectedAccessLevel
-      });
-
-      this.workspacesService.shareWorkspace(
+      const email = this.convertToEmail(this.toShare);
+      const role = this.selectedAccessLevel;
+      if (this.checkUnique(email, role)) {
+        // A user can only have one role on a workspace so we replace them in the list
+        const updateList = Array.from(this.workspace.userRoles)
+          .filter(function(r) { return r.email !== email; });
```

Comment:
Should use an ES6 arrow function (and can make it shorter with that): `.filter(r => r.email !== email);`

Reply:
Done